server/src/main/java/io/crate/analyze/AnalyzedTableElements.java

```diff
@@ -462,37 +489,46 @@ public static <T> void buildReference(RelationName relationName,
         boolean isNullable = !columnDefinition.hasNotNullConstraint();
-        if (isAddColumn && type.id() == GeoShapeType.ID) {
+        if (bound && type.id() == GeoShapeType.ID) {
             Map<String, Object> geoMap = new HashMap<>();
             if (columnDefinition.geoProperties() != null) {
-                // applySettings validates geo properties.
-                // If this method called from CreateTablePlan geoProperties are not yet resolved, they are still Literals. No need to validate, it will be done later.
-                // In case of ADD COLUMN we have to validate geo properties.
                 GeoSettingsApplier.applySettings(geoMap, (GenericProperties<Object>) columnDefinition.geoProperties(), columnDefinition.geoTree());
             }
             Float distError = (Float) geoMap.get("distance_error_pct");
+            // We need to use "all fields" constructor to make sure we cover all possible options when used in CREATE TABLE
             ref = new GeoReference(
-                columnDefinition.position,
                 new ReferenceIdent(relationName, columnDefinition.ident()),
-                isNullable,
+                RowGranularity.DOC,
                 realType,
+                ColumnPolicy.STRICT, // Use relevant for non-object field value.
+                IndexType.PLAIN,
+                isNullable,
+                columnDefinition.docValues(),
+                columnDefinition.position,
+                null,
+                columnDefinition.formattedDefaultExpression(),
                 columnDefinition.geoTree(),
                 (String) geoMap.get("precision"),
                 (Integer) geoMap.get("tree_levels"),
                 distError != null ? distError.doubleValue() : null
             );
-        } else if (isAddColumn && columnDefinition.analyzer() != null) {
-            // We are sending IndexReference since it's the only Reference implementation having 'analyzer'.
-            // However, IndexReference is dedicated to reflect index declaration like 'INDEX some_index using fulltext(some_col) with (analyzer = 'english')'
-            // Hence, we ignore copyTo which is irrelevant for ADD COLUMN.
-            // columnDefinition.isIndexColumn() cannot be used here, it's always false for ADD COLUMN.
+        } else if (bound && (columnDefinition.analyzer() != null || columnDefinition.copyTo().isEmpty() == false)) {
+            // If analyzer is not null, it's a column definition with inlined FULLTEXT.
+            // If columnDefinition.copyTo() is not empty, it's a column definition referred by a table level INDEX declaration.
+            // We need to use "all fields" constructor to make sure we cover all possible options when used in CREATE TABLE
             ref = new IndexReference(
-                columnDefinition.position,
+                new ReferenceIdent(relationName, columnDefinition.ident()),
+                RowGranularity.DOC,
+                realType,
```

Comment:
Need to pass realType in order not to lose Array - was a bug on master, merged https://github.com/crate/crate/commit/45516136db8565c33a0db38c5ea2205f53db9634